### DISCUSSION TITLE
Initial Delay & Other Questions

### DISCUSSION THREAD

**Discussion by bionemesis** [KelvinTegelaar/CIPP#24] [Q&A] [answered]
I've deployed this now a few times via Azure, mostly because I was convinced I did something wrong and then because I did do something wrong. What I'm finding is that after the initial deployment, when I follow the invite and login, my tenants are not displayed and I just have a "Loading..." screen. Yesterday, when I went through this and just kind of threw up my hands, I came back in the morning and all my tenants were there and it seemed to work. So my first question is, how long should it take before tenants show up? I know in the initial 20 minutes it's downloading things from Microsoft, and maybe those downloads contribute to the delay? Now, on this most recent deployment I have the new "Permissions Settings" which displays the following:

_We could not connect to the API to retrieve the permissions. There might be a problem with the secure application model configuration. The returned error is: 404 - Response status code does not indicate success: 404 (Not Found)._

What's interesting there is that the secure application model hasn't changed and is the same as the one used earlier today. Is it possible this is because those aforementioned modules are still downloading from Microsoft? And then my last question is around troubleshooting/logging. Is there a way to see what's currently going on to find out if there is some kind of error being generated?

**Comment by KelvinTegelaar** [comment, marked as answer]
good question actually! There's  a couple of factors at play here we've found.

The first is that the creation of the applications, and downloading of resources takes about 20 minutes, which is *really* annoying us so we're looking for a solution. 

The second is that it seems that the propagated permissions on the Azure side seem to take long too at times. You can see if this is an issue by going to your function app inside of the Azure portal and clicking on "configuration". If you see a red X next to the keyvault storage it means that the permissions have not applied yet.

I don't think we can really accelerate that, but I'll see if we can find a nice way to show an error or warning. It should be solved by just waiting....
